src/main/java/teammates/common/datatransfer/ResultFetchType.java

```diff
@@ -0,0 +1,34 @@
+package teammates.common.datatransfer;
+
+/**
+ * The result fetching type to indicate whether the fetch is for giver only, receiver only or both.
+ */
+public enum ResultFetchType {
+    /**
+     * Fetch by giver only.
+     */
+    GIVER_ONLY,
+    /**
+     * Fetch by receiver only.
+     */
+    RECEIVER_ONLY,
+    /**
+     * Fetch by both giver and receiver.
+     */
+    BOTH;
```

Comment:
Just a suggestion, perhaps you could add 2 boolean fields to this constructor, e.g. `fetchByGiver` and `fetchByReceiver`, and implement 2 methods `shouldFetchByGiver()` and `shouldFetchByReceiver()`. This is similar to `FeedbackQuestionType`.

This is so that later on, rather than implementing the same set of methods for both giver and receiver (e.g. `getFeedbackResponseCommentEntitiesForQuestionInReceiverSection`), you can just use a single method (e.g. `getFeedbackResponseCommentEntitiesForQuestion`) and modify its behaviour based on what is returned from `shouldFetchByGiver()` and `shouldFetchByReceiver()`.

Reply:
I see, this sounds quite extensible! I do have the same concern here but your way is much clearer :)
In some previous iterations of the review, @rrtheonlyone mentioned keeping things separated and hence there are only separate methods `getFeedbackResponseCommentEntitiesForQuestionInReceiverSection` and its giver counter part remaining; this was done intentionally after reverting a change that rope the two into one.

Reply:
Actually, after looking at some code segments, I think leaving the trichotomy as three different distinct unit methods (`InGiverSection`, `InReceiverSection`, `In(Both)Section` might be cleaner at the `Storage` layer, as then only `Logic` needs to figure out which one of the above methods to use to get the correct response needed.

Can clarify too if you meant to say `BOTH(Boolean fetchByGiver, Boolean fetchByReceiver)`?